Assets/Scripts/GameManager.cs

```diff
@@ -71,18 +96,25 @@ public class GameManager : MonoBehaviour {
             StopAllCoroutines();
             StartCoroutine(TypeSentence(CurrentQuestion.Question));
             for (var i = 0; i < CurrentQuestion.Answers.Length; i++) {
-                ButtonTexts[i].text = CurrentQuestion.Answers[i].Label;
-                AnswerButtons[i].gameObject.SetActive(true);
-            }
+                var CurrentAnswer = CurrentQuestion.Answers[i];                                    
+                bool activateButton = true;
+                var RequiredConditions = CurrentAnswer.Required ?? new Dictionary<string, object>();
 
-            if (!string.IsNullOrWhiteSpace(CurrentQuestion.Image)) {
-                Sprite background = Resources.Load<Sprite>(CurrentQuestion.Image);
-                BackgroundImage.sprite = background;
-            }else {
-                Sprite background = Resources.Load<Sprite>("Sprites/Backgrounds/black");
-                BackgroundImage.sprite = background;
-            }
+                foreach(var condition in RequiredConditions) {
+                    flags.TryGetValue(condition.Key, out object conditionValue);  
+                    activateButton = conditionValue.Equals(condition.Value);
+                    if(activateButton) break;
```

Comment:
Don't comment this. it should be (!activateButton) . Currently it works because we only have one condition, i'll change that when possible.